Switch to rkt

dm0- opened the pull request:
This fixes coreos/bugs#1610.

squeed commented on toolbox:
```diff
@@ -23,24 +20,42 @@ if [ -f "${toolboxrc}" ]; then
+
+exec sudo rkt --insecure-options=image run \
+    --interactive \
+    --mount=volume=hosts,target=/etc/hosts \
```
If you don't want the toobox to be able to edit /etc/hosts and /etc/resolv.conf, you can just run `rkt --hosts-entry=host --dns=host`

dm0- replied:
It seems that those options *do* allow editing the files, but I'm not aware of a requirement one way or the other.  I've changed the options since they are cleaner.

marineam commented:
My basic concern here is how this may impact existing toolbox users. Previously TOOLBOX_DIRECTORY, 
￼ TOOLBOX_BIND, and TOOLBOX_ENV could be customized in .toolboxrc. Previous state in TOOLBOX_DIRECTORY is now being ignored. Now, does state persist across invocations of toolbox like it did before? What trouble may be caused by nspawn previously using a new mount namespace but rkt fly not using one?

dm0- commented:
I've updated the commits with garbage collection as suggested by @lucab.

@marineam The `TOOLBOX_BIND` and `TOOLBOX_ENV` variables hold Docker-specific command arguments that would no longer be valid.  I've added the `TOOLBOX_RKT_ARGS` variable tho cover the functionality of both of them.  State does not persist with this, but I think the lack of a mount namespace was the entire reason for switching (see coreos/bugs#1610).

jonboulle commented on toolbox:
```diff
@@ -23,24 +20,42 @@ if [ -f "${toolboxrc}" ]; then
+    --mount=volume=run,target=/media/root/run \
+    --mount=volume=usr,target=/media/root/usr \
+    --net=host \
+    --stage1-path=/usr/lib/rkt/stage1-images/stage1-fly.aci \
```
is this caching the image now? otherwise there's a bit of a performance hit for repeated invocations.

marineam commented:
Loss of state between runs is going to really limit the utility of toolbox. I'm honestly not convinced that this is a thing worth doing, particularly since there's really no way to do this without breaking existing users of toolbox. There are behaviors of rkt that may suit some users better than toolbox but as the command line shows rkt isn't exactly an easy thing to use. Maybe instead of changing toolbox we add a new command `rktbox` to fill that niche?

dm0- commented:
From the `SIGKILL`s I've seen, this fixes coreos/bugs#1216.

dm0- commented:
This also fixes coreos/bugs#1704.

lucab commented on toolbox:
```diff
@@ -23,24 +20,45 @@ if [ -f "${toolboxrc}" ]; then
+    --interactive \
+    --mount=volume=root,target=/media/root \
+    --mount=volume=run,target=/media/root/run \
+    --mount=volume=usr,target=/media/root/usr \
```
Jumping here from https://github.com/coreos/bugs/issues/1630.

This is nesting `run` and `usr` under an existing `root` volume. Due to "shared:1" back propagation, they will re-appear back on the host layered on top the rootfs. That's an almost pathological case to clean, as they'll easily become busy. I strongly suggest to use separate non-nested targets for each volume and if possible not to share root at all (ie. enumerate single volume sources).

dm0- commented:
There are toolbox redesign plans in the works, so closing this ancient PR.
